operator/v1/types_machineconfiguration.go

```diff
@@ -56,3 +65,98 @@ type MachineConfigurationList struct {
 	// Items contains the items
 	Items []MachineConfiguration `json:"items"`
 }
+
+type ManagedBootImages struct {
+	// machineManagers is an array of machineManager objects.
+	// The MCO will watch for changes to this list and register/de-register machine management resources from boot image updates.
+	// An entry in this list consists of the resource type, the API group that the resource belongs to and a selection filter
+	// on the resources.
+	//
+	// Warning: Only one entry is permitted per unique pair of resource/API group. The label selector provided within MachineManager
+	// can be used for further customization if required.
+	//
+	// +optional
+	// +listType=map
+	// +listMapKey=resource
+	// +listMapKey=apiGroup
+	MachineManagers []MachineManager `json:"machineManagers"`
+}
+
+// MachineManager contains identifying information of a machine management resource(eg. a machineset) that will be
+// registered for boot image updates. This is likely to evolve as support for more machine management resources are added.
+type MachineManager struct {
+	// resource is the machine management resource's type.
+	//
+	// The following values are accepted:
+	// - MachineSets: The machine manager will only register resources of the type MachineSet, which may belong to MachineAPI or ClusterAPI.
+	//
+	// +kubebuilder:validation:Required
+	Resource MachineManagerMachineSetsResourceType `json:"resource"`
+	// apiGroup is name of the APIGroup that the machine management resource belongs to.
+	//
+	// The following values are accepted:
+	// - MachineAPI: The machine manager will only register resources that belong to MachineAPI APIGroup.
+	//
+	// +kubebuilder:validation:Required
+	APIGroup MachineManagerMachineSetsAPIGroupType `json:"apiGroup"`
+	// selection allows granular control of the machine management resources that will be registered for boot image updates.
+	//
+	// +kubebuilder:validation:Required
+	Selection MachineManagerSelector `json:"selection"`
+}
+
+// +kubebuilder:validation:XValidation:rule="has(self.mode) && self.mode == 'Partial' ?  has(self.partial) : !has(self.partial)",message="Partial is required when type is partial, and forbidden otherwise"
+// +union
+type MachineManagerSelector struct {
+	// mode is a union discriminator for MachineManagerSelector and can have three possible values.
```

Comment:
You're talking here in technical terms, you should talk more in end user terminology.
```suggestion
	// mode determines how machine managers will be selected for updates.
	// Valid values are All, None and Partial.
	// All means that every resource matched by the machine manager will be updated.
	// None means that no resources matched by the machine manager will be updated.
	// Partial requires a specified selector and allows customisation of which resources matched by the machine manager will be updated.
```
This got me thinking, what's the point to None? Surely you just turn off the feature completely?

Reply:
I'm not opposed to taking this out, but I figured this would give them an easy way to negate a MachineSet/group of MachineSets